### PR TITLE
FIX-#5150: Sync row labels after read_csv when index_col is False

### LINKED ISSUE #5150: BUG: Internal and external indices on axis 0 do not match doing `df._to_pandas` after `pd.read_csv`
### Modin version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest released version of Modin.

- [X] I have confirmed this bug exists on the main branch of Modin. (In order to do this you can follow [this guide](https://modin.readthedocs.io/en/stable/getting_started/installation.html#installing-from-the-github-master-branch).)

### Reproducible Example

```python
import numpy as np
import pandas
import modin.pandas as pd

pandas_df = pandas.DataFrame(np.random.randint(0, 100, size=(2**6, 2**6)))
pandas_df.to_csv("foo.csv", index=False)
modin_df = pd.read_csv("foo.csv", index_col=False)
pandas_df2 = modin_df._to_pandas()
```

### Issue Description

Internal and external indices on axis 0 do not match doing `df._to_pandas` after `pd.read_csv`. The indices should be syncronized after `read_csv` in the case.

### Expected Behavior

Modin should correctly read data with the provided parameters so consequent operations work correctly too.

### Error Logs

<details>

```python-traceback

Exception: Internal Error. Please visit https://github.com/modin-project/modin/issues to file an issue with the traceback and the command that caused this error. If you can't file a GitHub issue, please email bug_reports@modin.org.
Internal and external indices on axis 0 do not match.

```

</details>

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 6cc441a44c5a6026287818b6c9dc25e3b3342037
python           : 3.9.13.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.10.102.1-microsoft-standard-WSL2
Version          : #1 SMP Wed Mar 2 00:30:59 UTC 2022
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

Modin dependencies
------------------
modin            : 0.16.0+21.g6cc441a4
ray              : 2.0.0
dask             : 2022.9.1
distributed      : 2022.9.1
hdk              : None

pandas dependencies
-------------------
pandas           : 1.5.1
numpy            : 1.23.3
pytz             : 2022.2.1
dateutil         : 2.8.2
setuptools       : 65.4.0
pip              : 22.2.2
Cython           : None
pytest           : 7.1.3
hypothesis       : None
sphinx           : 5.2.2
blosc            : None
feather          : 0.4.1
xlsxwriter       : None
lxml.etree       : 4.8.0
html5lib         : None
pymysql          : None
psycopg2         : 2.8.6
jinja2           : 3.1.2
IPython          : 8.5.0
pandas_datareader: None
bs4              : 4.11.1
bottleneck       : None
brotli           :
fastparquet      : 0.8.3
fsspec           : 2022.8.2
gcsfs            : None
matplotlib       : 2.2.5
numba            : None
numexpr          : 2.8.3
odfpy            : None
openpyxl         : 3.0.10
pandas_gbq       : 0.17.8
pyarrow          : 9.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : 2022.8.2
scipy            : 1.9.1
snappy           : None
sqlalchemy       : 1.4.41
tables           : 3.7.0
tabulate         : None
xarray           : 2022.6.0
xlrd             : 2.0.1
xlwt             : None
zstandard        : None
tzdata           : None

</details>

### PR DESCRIPTION
**Pull request by YarShev**
Signed-off-by: Igoshev, Iaroslav <iaroslav.igoshev@intel.com>

## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5150  
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/test/test_io.py [hunk 1]
```diff
@@ -1334,6 +1334,14 @@ def test_to_csv_with_index(self):
         ).set_index("key")
         eval_to_file(modin_df, pandas_df, "to_csv", "csv")
 
+    def test_read_csv_issue_5150(self):
+        pandas_df = pandas.DataFrame(np.random.randint(0, 100, size=(2**6, 2**6)))
+        pandas_df.to_csv("issue5150.csv", index=False)
```
Comment (dchigarev) [inline modin/pandas/test/test_io.py:1339]: this creates a new file and doesn't clean it after the test is completed, right? It seems that the issue doesn't depend on that concrete dataset so we could use any other of the existing CSV files in `modin/pandas/test/data/`? 

If you'd still like to create a new file in the test, I would suggest doing it at least under `ensure_clean()` context or using one of our IO test functions that can create&clean test files.

Comment (YarShev) [reply modin/pandas/test/test_io.py]: fixed

**Review by dchigarev — APPROVED** [review]
LGTM!

File: modin/pandas/test/test_io.py [hunk 2]
```diff
@@ -1334,6 +1334,17 @@ def test_to_csv_with_index(self):
...
+    def test_read_csv_issue_5150(self):
+        with ensure_clean(".csv") as unique_filename:
+            pandas_df = pandas.DataFrame(
+                np.random.randint(0, 100, size=(2**6, 2**6))
+            )
+            pandas_df.to_csv(unique_filename, index=False)
+            expected_pandas_df = pandas.read_csv(unique_filename, index_col=False)
+            modin_df = pd.read_csv(unique_filename, index_col=False)
+            actual_pandas_df = modin_df._to_pandas()
+        df_equals(expected_pandas_df, actual_pandas_df)
```
Comment (pyrito) [inline modin/pandas/test/test_io.py:1346]: Is it possible to use `eval_io` here instead?

**Review by pyrito — COMMENTED** [review]
LGTM overall just a style comment

**Review by vnlitvinov — APPROVED** [review]

**Pull request merged**